src/components/base/outline-google-map/outline-google-map.ts

```diff
@@ -0,0 +1,248 @@
+import { LitElement, html, CSSResultGroup } from 'lit';
+import { customElement, property, query } from 'lit/decorators.js';
+import { GoogleMapsApi } from './src/google-maps-api';
+import './src/google-maps-api';
+import { OutlineMapMarker } from './src/outline-map-marker';
+import { MapSelector } from './src/map-selector';
+import './src/map-selector';
+import componentStyles from './outline-google-map.css.lit';
+
+export type MapType = 'roadmap' | 'hybrid' | 'satellite' | 'terrain';
+export type GestureType = 'auto' | 'cooperative' | 'greedy' | 'none';
+
+@customElement('outline-google-map')
+export class OutlineGoogleMap extends LitElement {
+  static styles: CSSResultGroup = [componentStyles];
+  /**
+   * A Maps API key. To obtain an API key, see https://developers.google.com/maps/documentation/javascript/tutorial#api_key.
+   */
+  @property({ type: String })
+  apiKey = '';
```

Comment:
Minor: We all have different opinions, but I recommend always using the options below:

`{ type: ..., reflect: true, attribute: ... }`

so

`{ type: String, reflect: true, attribute: 'api-key' }`

The reasoning:
- Consistent kabob case for attributes
- Reflection makes the developer inspection match the component state if you modify it in some way. Otherwise you can end up in situations where someone looks at the inspector and the state and attributes have diverged and that seems unnecessarily confusing.